pkg/acs/calacs/acsrej/acsrej_loop.c

```diff
@@ -403,304 +463,318 @@ int acsrej_loop (IODescPtr ipsci[], IODescPtr ipdq[],
+
                                     if (mask[n][jj][ii] != HIT)
                                         mask[n][jj][ii] = SPILL;
                                 }
                             }
                         }
-                    } /* End of loop over i */
+                    } /* End of loop over both amps used for line */
 
                     /* accumulate the total counts in each good pixel */
                     for (i = 0; i < dim_x; i++)  {
-                        if ( (mask[n][width][i] & maskdq) == OK ) {
+                        if ( (mask[n][width][i] & maskdq ) == OK ) {
                             /* add the sky-subtracted but UN-scaled counts */
-                            sum[i] += pic[n][width][i] * efacn;
-                            PIX(efacsum,i,line,dim_x) += efacn;
+                            sum[i] += pic[n][width][i] * efacn;  /* e */
+                            PIX(efacsum, i, line, dim_x) += efacn;  /* s */
                         }
-                    } /* End loop over i (X) */
+                    }
+
+                    /* On the last iteration, accumulate variance and DQ vals.
 
-                    /* On the last iteration, accumulate variance and DQ vals */
+                       Accumulate the variance only during the last iteration
+                       and ONLY for non-HIT or non-SPILL pixels.
+                    */
                     if (iter == (niter-1)) {
 
-                        /* accumulate the variance only during the
-                           last iteration and ONLY for non-HIT or non-SPILL
-                           pixels. */
+                        /* FIRST AMP */
                         for (i = 0; i < ampx; i++) {
-                            if ( (mask[n][width][i] & maskdq) == OK) {
-                                dum = pic[n][width][i]*efacn + skyvaln;
+                            if ( (mask[n][width][i] & maskdq) == OK ) {
+                                /* e, sky added back in */
+                                dum = (pic[n][width][i] * efacn) + skyvaln;
 
                                 /* clip the data at zero */
-                                val = (dum > 0.) ? dum : 0.;
-
-                                /* Removed SCALENSE from error. 24-Nov-2003 WJH.
-                                sumvar[i] += nse[0] + val/gn[0] + SQ(scale*val);
-                                */
+                                val = (dum > 0.) ? dum : 0.;  /* e */
 
                                 if (newbias == 0) {
-                                    sumvar[i] += nse[0] + val/gn[0];
+                                    /* Variance term like DHB without SCALENSE
+                  https://trac.stsci.edu/ssb/stsci_python/ticket/1223#comment:13
+                                nse is e^2
+                                val is e (with sky)
+                                    */
+                                    sumvar[i] += nse[0] + val;
```

Comment:
Note to self: This computation is wrong and has been for some time.  After tracing each factor in the equation to make sure of the units, the comment is correct.  nse is electrons2, but val is on electrons.

Reply:
Need to double-check this as the comments are confusing.

Reply:
Will address this in Issue#293.